Исправление моей части task_2

KashKetchup opened the pull request.

KashKetchup committed:
Исправление неправильного массива доски

KashKetchup committed:
Merge remote-tracking branch 'origin/task_2' into task_2

# Conflicts:
#	src/main/java/org/example/GameState.java

KashKetchup committed:
Добавление класса ChessmenMovement

KashKetchup committed:
Измение логики Bishop,Rook и Queen. Queen больше не зависит от Bishop и rook

KashKetchup committed:
Исправления ошибок в task_2, добавление тестов для

KashKetchup committed:
Merge branch 'task_2' into task_2_movement

AlexV0rob commented on src/main/java/org/example/chess/Bishop.java:
```diff
@@ -1,175 +1,39 @@
          * Проверяем правильность хода в два этапа:
-         * 1) Смотрим что интересующая насклетка нас не занята или там находится
+         * 1) Смотрим что интересующая нас етка нас не занята или там находится
          * шахматная фигура оппонента
```
Весь комментарий теперь лишний

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/chess/Bishop.java:
```diff
@@ -1,175 +1,39 @@
-            if (Math.abs(start.row() - finish.row()) == Math.abs(start.column() - finish.column()) 
-            		&& isWayFree(start, finish, board)) {
+        if (board[finish.row()][finish.column()] == 0 ||
+                board[finish.row()][finish.column()] * board[start.row()][start.column()] < 0) {
```
А чего от методов isPositionEmpty и isPositionEnemy отказался?

KashKetchup replied:
Они приватные теперь

AlexV0rob replied:
Так можно же сделать собственные методы в этом классе, как сделано, например, в пешке

KashKetchup replied:
А стоит? Ради единичного использования

AlexV0rob replied:
Код более читаемый

KashKetchup replied:
Хорошо, сейчас сделаю

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/chess/Rook.java:
```diff
@@ -8,162 +10,28 @@
     @Override
     public boolean checkMove(PositionOnBoard start, PositionOnBoard finish, byte[][] board) {
         if (board[finish.row()][finish.column()] == 0 ||
                 (board[finish.row()][finish.column()] < 0 != board[start.row()][start.column()] < 0)) {
```
Здесь тоже можно сделать методы isPositionEmpty и isPositionEnemy

KashKetchup replied:
Ответил выше

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/movement/ChessmenMovement.java:
```diff
@@ -0,0 +1,198 @@
+    /**
+     * Делает список доступых ходов в заданном направлении
+     */
+    public List<PositionOnBoard> allVerticalAndHorizontalmoves(PositionOnBoard start, byte[][] board) {
```
moves лучше с большой буквы

KashKetchup replied:
Странно, я точно помню что делал Рефакторинг названия...

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/movement/ChessmenMovement.java:
```diff
@@ -0,0 +1,198 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class ChessmenMovement {
```
JavaDoc

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/movement/ChessmenMovement.java:
```diff
@@ -0,0 +1,198 @@
+        int currentVerticalShift = 1;
+        int currentHorizontalShift = 1;
+        PositionOnBoard currentPosition = startPosition;
+        while (currentPosition != SIGNAL_POSITION &&
```
Может, лучше сравнивать поля позиции, он точно не будет здесь сравнивать ссылки?

KashKetchup replied:
Понял. Тебе устроит проверка через equals?

AlexV0rob replied:
Учитывая, что по умолчанию она сравнивает объект с самим собой (причём через равно), нет

KashKetchup replied:
Хорошо, я отдельно буду сравнивать строку и столбец

AlexV0rob commented on src/main/java/org/example/movement/ChessmenMovement.java:
```diff
@@ -0,0 +1,198 @@
+    }
+
+    /**
+     * Функция делающая сдвиг в заданном направлении
```
Пунктуация

KashKetchup replied:
Косяк, признаю

KashKetchup replied:
Исправлено

AlexV0rob commented on src/test/java/org/example/movement/ChessmenMovementTest.java:
```diff
@@ -0,0 +1,98 @@
+ * Тестирование методов классаChessmenMovementTest
+ */
+public class ChessmenMovementTest {
+    private final ChessmenMovement chessmenMovement = new ChessmenMovement();
```
JavaDoc

KashKetchup replied:
Не совсем понял

AlexV0rob replied:
На все поля нужно тоже делать JavaDoc, у тебя поле chessmenMovement без JavaDoc

KashKetchup replied:
Понял. Исправлено

AlexV0rob commented on src/test/java/org/example/movement/ChessmenMovementTest.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    /**
+     * Тестирование проверки отсуствия препятсвий на пути
```
В "препятствий" опечатка

KashKetchup replied:
Не только в нём

KashKetchup replied:
Исправлено

KashKetchup committed:
Исправление замечаний

KashKetchup committed:
Исправление замечаний

KashKetchup committed:
Merge remote-tracking branch 'origin/task_2_movement' into task_2_movement

KashKetchup committed:
Исправление замечаний, часть 2:
Исправление некорректного сравнения в ChessmenMovement, добавление JavaDoc'ов и тд

AlexV0rob commented on src/main/java/org/example/chess/Bishop.java:
```diff
@@ -37,139 +23,42 @@ private enum PositionRelatives {
-            currentColumn = nextPosition(currentColumn, horizontalRelatives);
-        }
-        return (currentRow == finish.row() && currentColumn == finish.column());
+    private boolean isInsideBorders(int pos) {
```
А зачем этот метод, проверка начальных и конечных координат на нахождение внутри доски есть в GameHandler

KashKetchup replied:
Признаю, не учёл этого

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/chess/Queen.java:
```diff
@@ -1,32 +1,71 @@
+    /**
+     * Проверить, что координата находится в границах
+     */
+    private boolean isInsideBorders(int pos) {
```
То же самое

KashKetchup replied:
Исправлено

AlexV0rob commented on src/main/java/org/example/chess/Rook.java:
```diff
@@ -36,134 +23,44 @@ private enum PositionRelatives {
-            currentColumn = nextPosition(currentColumn, horizontalRelatives);
-        }
-        return (currentRow == finish.row() && currentColumn == finish.column());
+    private boolean isInsideBorders(int pos) {
```
И здесь

KashKetchup replied:
Исправлено

KashKetchup committed:
Исправление замечаний, часть 3:
Удаление ненужного метода IsInsideBorders в Rook Bishop и Queen

AlexV0rob reviewed (approved):
Всё устраивает, можешь объединять
